d2core/d2input/keyboard/keyboard.go

```diff
@@ -0,0 +1,209 @@
+package keyboard
+
+type Key int
+
+const (
+	Key0 = Key(iota)
+	Key1
+	Key2
+	Key3
+	Key4
+	Key5
+	Key6
+	Key7
+	Key8
+	Key9
+	KeyA
+	KeyB
+	KeyC
+	KeyD
+	KeyE
+	KeyF
+	KeyG
+	KeyH
+	KeyI
+	KeyJ
+	KeyK
+	KeyL
+	KeyM
+	KeyN
+	KeyO
+	KeyP
+	KeyQ
+	KeyR
+	KeyS
+	KeyT
+	KeyU
+	KeyV
+	KeyW
+	KeyX
+	KeyY
+	KeyZ
+	KeyApostrophe
+	KeyBackslash
+	KeyBackspace
+	KeyCapsLock
+	KeyComma
+	KeyDelete
+	KeyDown
+	KeyEnd
+	KeyEnter
+	KeyEqual
+	KeyEscape
+	KeyF1
+	KeyF2
+	KeyF3
+	KeyF4
+	KeyF5
+	KeyF6
+	KeyF7
+	KeyF8
+	KeyF9
+	KeyF10
+	KeyF11
+	KeyF12
+	KeyGraveAccent
+	KeyHome
+	KeyInsert
+	KeyKP0
+	KeyKP1
+	KeyKP2
+	KeyKP3
+	KeyKP4
+	KeyKP5
+	KeyKP6
+	KeyKP7
+	KeyKP8
+	KeyKP9
+	KeyKPAdd
+	KeyKPDecimal
+	KeyKPDivide
+	KeyKPEnter
+	KeyKPEqual
+	KeyKPMultiply
+	KeyKPSubtract
+	KeyLeft
+	KeyLeftBracket
+	KeyMenu
+	KeyMinus
+	KeyNumLock
+	KeyPageDown
+	KeyPageUp
+	KeyPause
+	KeyPeriod
+	KeyPrintScreen
+	KeyRight
+	KeyRightBracket
+	KeyScrollLock
+	KeySemicolon
+	KeySlash
+	KeySpace
+	KeyTab
+	KeyUp
+	KeyAlt
+	KeyControl
+	KeyShift
+)
+
+var Keys = [...]Key{
```

Comment:
Don't use variables for things that are actually supposed to be constants.

Reply:
I don't thing the array can be const. It's just there so we can range over the keys elsewhere. How should it be done as a const?

Reply:
Okay, I've addressed this in my recent commits.